Dockerfile.sdk

```diff
@@ -57,6 +57,7 @@ FROM ${BASE_IMAGE} AS sdk_build
 
 # Ensure apt-get won't prompt for selecting options
 ENV DEBIAN_FRONTEND=noninteractive
+ENV PIP_BREAK_SYSTEM_PACKAGES=1
```

Comment:
>--break-system-packages
>    Allow pip to modify an EXTERNALLY-MANAGED Python installation

Could you please clarify which package require it?

Reply:
All non virtual env installation via pip require this flag now - https://stackoverflow.com/questions/75602063/pip-install-r-requirements-txt-is-failing-this-environment-is-externally-mana

I have added this flag for now to avoid failure down the stack and avoid updating all pip commands. We would need to evaluate switching to python venv's in future for pip install commands

Reply:
I see that link you are referring to is related to https://peps.python.org/pep-0668/ which is outdated. 
Could you please elaborate your decision.

Reply:
Link further points to updated links i think. Anyway, pip has a behaviour change wrt to all system wide installation (non virtual env install). Python base system environment is considered as externally managed and fails to install any packages without this flag. Recommendation from pip side is to either use OS package manager to install such packages on system level, use new pipx tool which behind the scene creates a temporary venv and installs things there (mainly targeted for any CLI python tools) or switch to a virtual env install. You can read the full error message on above stackoverflow link.

This `PIP_BREAK_SYSTEM_PACKAGES` flag allows us to override this new behavior and continue running pip install commands without any venv activated. However, it still wont allow any upgrades to debian installed packags like python3-pip, wheel, setuptools etc. So i switched whatever packages are available as debian install to install via apt and rest pip install commands will utilize this flag to continue for now with base installation.

Reply:
Then logical way here should be usage of virtual environment otherwise we are mixing up system and user space.